From 7b276b5eeb7bb84a872354fbe1fc1c87ab65ecf4 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Thu, 27 Oct 2016 18:50:41 +0200
Subject: [PATCH 2/7] libqmi-glib,proxy: print error message on proxy open
 response error, and plug memleak

(cherry picked from commit 237d8c38ff2977b53de0dd961fadeb857d7b354f)
---
 src/libqmi-glib/qmi-proxy.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libqmi-glib/qmi-proxy.c b/src/libqmi-glib/qmi-proxy.c
index c5e936e..36c85b5 100644
--- a/src/libqmi-glib/qmi-proxy.c
+++ b/src/libqmi-glib/qmi-proxy.c
@@ -251,6 +251,8 @@ complete_internal_proxy_open (Client *client)
     response = qmi_message_response_new (client->internal_proxy_open_request, QMI_PROTOCOL_ERROR_NONE);
 
     if (!send_message (client, response, &error)) {
+        g_warning ("couldn't send proxy open response to client: %s", error->message);
+        g_error_free (error);
         qmi_message_unref (response);
         connection_close (client);
         return;
-- 
2.10.1

